db/src/utils.rs

```diff
@@ -0,0 +1,35 @@
+use anyhow::{Error, Result};
+use sqlx::any::{Any, AnyPool, AnyPoolOptions};
+use sqlx::migrate;
+use sqlx::migrate::MigrateDatabase;
+
+/// Re-export of generic connection pool type.
+pub type Pool = AnyPool;
+
+/// Create database when not existing.
+pub async fn create_database(url: String) -> Result<()> {
+    if !Any::database_exists(&url).await? {
+        Any::create_database(&url).await?;
+    }
+
+    Ok(())
+}
+
+/// Create a database agnostic connection pool.
+pub async fn connection_pool(
+    url: String,
+    max_connections: u32,
+) -> std::result::Result<Pool, Error> {
+    let pool: Pool = AnyPoolOptions::new()
+        .max_connections(max_connections)
+        .connect(&url)
+        .await?;
+
+    Ok(pool)
+}
+
+/// Run any pending database migrations from inside the application.
+pub async fn run_pending_migrations(pool: &Pool) -> Result<()> {
+    migrate!("./migrations").run(pool).await?;
```

Comment:
Don't we need the `migrate` cargo feature flag for this?

https://github.com/launchbadge/sqlx#cargo-feature-flags

Reply:
Could also be interesting for migration management: https://crates.io/crates/sqlx-cli

Reply:
Also: where is this "./migrations" folder?

Reply:
> Don't we need the `migrate` cargo feature flag for this?
> 
> https://github.com/launchbadge/sqlx#cargo-feature-flags

It is set by default 🙂

Reply:
> Also: where is this "./migrations" folder?

Its in the `db` folder but not checked in as it's empty, will add a `.gitkeep` 👍